modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/QuantumToolkit.java

```diff
@@ -1497,44 +1553,95 @@ public void run() {
                 }
 
                 boolean errored = false;
+                // A temp QuantumImage used only as a RTT cache for rendering tiles.
+                var tileRttCache = new QuantumImage((com.sun.prism.Image) null);
```

Comment:
In the case of a snapshot that isn't tiled, you will end up creating and disposing a dummy `QuantumImage` that is never used. Maybe it would be worth initializing it to null here, constructing the object in the `if (h > maxTextureSize || w > maxTextureSize)` block, and checking for null before disposing in the finalize?